src/System.Diagnostics.StackTrace.Symbols/src/System/Diagnostics/StackTrace/Symbols.cs

```diff
@@ -0,0 +1,204 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+
+namespace System.Diagnostics.StackTrace
```

Comment:
I created a separate assembly because it was supposed to be optional since it references System.Reflection.Metadata, System.IO.FileSystem, etc.  This would allow us to decide separately on the “policy” of when to pay for overhead of display source/line info.  My current thinking was to add it to the shared runtime packages since System.Reflection.Metadata is already there, but allow developers to decide if they want the overhead for their standalone projects.  FYI, on Windows the source/line info is conditional too based on if msdia is installed.

Adding it to the existing System.Diagnostics.StackTrace is ok with me if everybody is ok with overhead. I don’t have a good sense on the magnitude of this overhead. I may be making a bigger deal of this that necessary.

From: Jan Kotas [mailto:notifications@github.com]
Sent: Tuesday, May 17, 2016 7:35 AM
To: dotnet/corefx <corefx@noreply.github.com>
Cc: Mike McLaughlin <mikem@microsoft.com>; Assign <assign@noreply.github.com>
Subject: Re: [dotnet/corefx] Add symbol support helper assembly for mscorlib's StackTrace. (#8591)


In src/System.Diagnostics.StackTrace.Symbols/src/System/Diagnostics/StackTrace/Symbols.cs<https://github.com/dotnet/corefx/pull/8591#discussion_r63533156>:

> @@ -0,0 +1,204 @@

> +// Licensed to the .NET Foundation under one or more agreements.

> +// The .NET Foundation licenses this file to you under the MIT license.

> +// See the LICENSE file in the project root for more information.

> +

> +using System.Collections.Immutable;

> +using System.IO;

> +using System.Reflection.Metadata;

> +using System.Reflection.Metadata.Ecma335;

> +using System.Reflection.PortableExecutable;

> +

> +namespace System.Diagnostics.StackTrace

Have we considered putting it in System.Diagnostics.StackTrace assembly instead?

I agree that System.Diagnostics.StackTrace may be a better home for this.

—
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corefx/pull/8591/files/f9e28947131c090d161d162af89f64720604d116#r63533156>